app/helpers/organisations_helper.rb

```diff
@@ -37,4 +38,14 @@ def organisation_name_row(user:, organisation:, summary_list:)
       end
     end
   end
+
+  def rent_periods_with_checked_attr(checked_periods: nil)
+    all_rent_periods = RentPeriod.rent_period_mappings
+    rent_periods = all_rent_periods.map do |code, period|
+      period_copy = period.clone
+      period_copy[:checked] = true if checked_periods.nil? || checked_periods.include?(code)
+      [code, period_copy]
+    end
+    rent_periods.to_h
+  end
```

Comment:
Would this do the same thing?
```
  def rent_periods_with_checked_attr(checked_periods: nil)
    all_rent_periods = RentPeriod.rent_period_mappings
    rent_periods = all_rent_periods.each_with_object({}) do |(code, period), hash|
      hash[code] = period.merge(checked: checked_periods.nil? || checked_periods.include?(code))
    end
  end
  ```

Reply:
I was looking for an enumerator that returned a hash. thanks

Reply:
then I htought I would try with inject because that sounds even more like what we're doing and the linter told me to do what you did and explained why. all good learning.